controllers/healthcheck_controller.go

```diff
@@ -673,7 +675,9 @@ func (r *HealthCheckReconciler) watchWorkflowReschedule(ctx context.Context, req
 		}
 		// reschedule next run of workflow
 		helper := r.createSubmitWorkflowHelper(ctx, log, wfNamespace, hc)
+		r.Locker.Lock()
 		r.RepeatTimersByName[hc.GetName()] = time.AfterFunc(time.Duration(repeatAfterSec)*time.Second, helper)
```

Comment:
Are there any reads we make from `r.RepeatTimersByName`? do we need any more read locking here?

Reply:
You may want to change to a RWMutex, and add Read locks in block such as :
https://github.com/keikoproj/active-monitor/blob/d692782d2efa2f58b996c89cb67260c140a290a3/controllers/healthcheck_controller.go#L131-L136

Also might want to check for key's existence before getting it, you can probably create a `GetTimerByName()` method that does both

Reply:
Looks like line 134 above is also a read operation, and is not thread safe, right?
`r.RepeatTimersByName[req.NamespacedName.Name].Stop()`

if `GetTimerByName() ` returns a pointer while using a lock it should be safe

Reply:
Thanks for the input.. yeah as they are referred by multiple go routines.. we should make it read lock as well..